partner/app/main.py

```diff
@@ -89,6 +104,21 @@ async def read_tilesp(
     results: int = Query(1, example=2),
 ):
     """Endpoint for requests from Contile."""
+    unknown_params = []
+    for param in request.query_params:
+        if param not in ACCEPTED_QUERY_PARAMS:
+            unknown_params.append(param)
+
+    if unknown_params:
+        return JSONResponse(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            content={"unaccepted query parameter": unknown_params},
```

Comment:
The response also needs to contain the fields from the API specification so that Contile can properly parse that error. See `body_from_API_spec` in `validation_exception_handler`. Feel free to move that dict to the module scope as a constant and use that in both places.

Reply:
I was able to incorporate all the other comments above. I am having trouble understanding this one - as in what's expected from your comments above. Can I use the same `validation_exception_handler` and it's `body_from_API_spec` to be used here?

Reply:
You will need to include the following in the response content, because according to the API spec a response with an error contains these fields:

```Python
body_from_API_spec = {
    "status": {"code": "103", "text": "Invalid input"},
    "count": "0",
    "response": "1",
}
```

This is currently a local variable in `validation_exception_handler`. I suggest to move this outside of that function to the module scope and use that in both places. In `validation_exception_handler` and `read_tilesp`:

For example:

```python
    if unknown_params:
        return JSONResponse(
            status_code=status.HTTP_400_BAD_REQUEST,
            content=jsonable_encoder(
                {
                    "test": {"unaccepted query parameter": unknown_params},
                    **BODY_FROM_API_SPEC,
                }
            ),
        )
```

The `**` syntax means that the dict will be updated with the items from `BODY_FROM_API_SPEC`.